wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -86,48 +74,55 @@
 	 * Name of the HTTP parameter to submit the requested result format to the
 	 * API.
 	 */
-	public final static String PARAM_FORMAT = "format";
+	protected final static String PARAM_FORMAT = "format";
 
 	/**
-	 * MediaWiki assert= parameter to ensure we are editting while logged in.
+	 * MediaWiki assert parameter to ensure we are editing while logged in.
 	 */
-	private static final String ASSERT_PARAMETER = "assert";
+	protected static final String ASSERT_PARAMETER = "assert";
+
+	protected static final MediaType URLENCODED_MEDIA_TYPE = MediaType.parse("application/x-www-form-urlencoded");
 
 	/**
 	 * URL to access the Wikibase API.
 	 */
-	final String apiBaseUrl;
+	protected final String apiBaseUrl;
 
 	/**
 	 * True after successful login.
 	 */
-	boolean loggedIn = false;
+	protected boolean loggedIn = false;
 	/**
 	 * User name used to log in.
 	 */
-	String username = "";
-	
+	protected String username = "";
```

Comment:
Maybe dumb question: is this field actually useful for the OAuth authentification? If it is the case, what about moving it to the `BasicApiConnection`?

Reply:
It's not useful for the authentification, but it's useful for getting the username via the OAuth credentials. You can see the overridden getCurrentUser() method actually uses it.